Sources/Meow/Model.swift

```diff
@@ -74,8 +74,10 @@ extension MutableModel {
     
     public func save(in database: MeowDatabase) -> EventLoopFuture<MeowOperationResult> {
         return database.collection(for: Self.self).upsert(self).map { reply in
+            let updateSuccess = reply.updatedCount == 1 && reply.upserted == nil
+            let upsertSuccess = reply.updatedCount == 0 && reply.upserted != nil
```

Comment:
I don't feel comfortable wit these conditions. I know it can't happen, but this code would fail when `updatedCount > 1` or `reply.updatedCount != 0 && reply.upserted != nil`. It feels very error prone, and we're really only interested in the existance of this insert. It would probably be simpler to read if we said r`eply.updatedCount > 0 || reply.upserted != nil`